orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/PipelineStarterListener.groovy

```diff
@@ -50,41 +40,6 @@ class PipelineStarterListener implements JobExecutionListener {
 
   @Override
   void afterJob(JobExecution jobExecution) {
-    startTracker.getAllStartedExecutions().each { startedExecutionId ->
-      try {
-        def execution = executionRepository.retrievePipeline(startedExecutionId)
-        if (execution.status.complete) {
-          processPipelines(execution)
-        }
-      } catch (ExecutionNotFoundException ignored) {
-        log.warn("Unable to update pipeline status for missing execution (executionId: ${startedExecutionId})")
-      } catch (Exception e) {
-        log.error('failed to update pipeline status', e)
-      }
-    }
-  }
-
-  void processPipelines(Pipeline execution) {
-    log.info("marking pipeline finished ${execution.id}")
-    startTracker.markAsFinished(execution.pipelineConfigId, execution.id)
-    if (execution.pipelineConfigId) {
-      def queuedPipelines = startTracker.getQueuedPipelines(execution.pipelineConfigId)
-      if (!queuedPipelines.empty) {
-        def nextPipelineId = queuedPipelines.first()
-        queuedPipelines.each { id ->
-          if (id == nextPipelineId) {
-            def queuedExecution = executionRepository.retrievePipeline(id)
-            log.info("starting pipeline ${nextPipelineId} due to ${execution.id} ending")
-            pipelineStarter.startExecution(queuedExecution)
-            startTracker.removeFromQueue(execution.pipelineConfigId, id)
-          } else if (!execution.keepWaitingPipelines) {
-            log.info("marking pipeline ${nextPipelineId} as canceled due to ${execution.id} ending")
-            executionRepository.cancel(id)
-            startTracker.removeFromQueue(execution.pipelineConfigId, id)
-          } // else we want to keep the pipeline in the queue
-
-        }
-      }
-    }
+    startTracker.triggerQueuedPipelines()
```

Comment:
All this logic is moved to `PipelineStartTracker` so it can be called outside of the context of a listener.